### PR TITLE
Backport 2.28: Fix IAR warnings

### PR DESCRIPTION
**Pull request by paul-elliott-arm**
## Description

IAR was warning that conditional execution could bypass initialisation of variables, although those same variables were not used uninitialised, however this is a quick fix to keep IAR quiet.

This will need forward porting to development once it has been checked to be ok (I cannot run IAR here)

## Gatekeeper checklist

- [x] **changelog** provided
- [X] **forward-port** #7090 (See comment about #6754)
- [x] **tests** ~~provided, or~~ not required

### CODE REVIEW

**Comment by tom-cosgrove-arm** [comment]
When you say "forward port done", are you referring to #6754 which includes the `exponent_bits_in_window` fix?

**Comment by paul-elliott-arm** [comment]
Ah no, I meant it is yet to be done, and I hadn't yet spotted that, thank you.

**Review by daverodgman — DISMISSED** [review]

**Comment by daverodgman** [comment]
> should this have a `ChangeLog` entry? We do have entries such as `Fix compilation warnings with IAR toolchain`, and might be important to users of the library...

Agreed, but it should be integrated into the existing Changelog addition from #6888, rather than have separate bullet points for each warning fix.

**Comment by Kasimashi** [comment]
Some warning are remaining please can you fix it before merging ? : 
#6924 

;)

**Comment by paul-elliott-arm** [comment]
Added new requested fixes, and added an edit to the previous IAR changelog. @Kasimashi - please re-review, and could you also take a look at https://github.com/Mbed-TLS/mbedtls/pull/6754 and confirm that we do not need any of those warning fixes in 2.28.2

File: ChangeLog.d/fix-iar-warnings.txt
Comment (tom-cosgrove-arm) [inline ChangeLog.d/fix-iar-warnings.txt:3]: (minor) I think we normally just say

```
   * Fix IAR compiler warnings. Contributed by Glenn Strauss in #3835.
     Fixes #6924.
```

Comment (gilles-peskine-arm) [reply ChangeLog.d/fix-iar-warnings.txt:3]: It's the other way round, actually: we do, in principle, credit external reporters in changelog entries. Always for security issues, not always for ordinary bugs. We generally don't credit contributors these days: it's understood that the credit is in the author fields in the git log.

Comment (tom-cosgrove-arm) [reply ChangeLog.d/fix-iar-warnings.txt:3]: However, in this case the fix was only given in a comment on an issue, and so Glenn is not the author of the commit, and #6924 is not a security issue, but the reporting of warnings issued by a compiler. So yes "I think we normally just say" maybe should be "In this case I think we should just say"

**Review by gilles-peskine-arm — DISMISSED** [review]
Approved — I don't know if this fixes all IAR warnings, but at least it's non-harmful.

**Comment by gilles-peskine-arm** [comment]
The code style check is complaining.

Also, what's the status with forward-porting?

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM

**Review by daverodgman — DISMISSED** [review]

**Comment by paul-elliott-arm** [comment]
Rebased, and hopefully code-styled.

**Comment by paul-elliott-arm** [comment]
Have provided my own forward port in order to get this one through (https://github.com/Mbed-TLS/mbedtls/pull/6754 contains much other work, which will also require backports, and this will muddy the water somewhat). I suggest we look at #6754 next.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by tom-cosgrove-arm** [comment]
CI is fully green

**Review by gilles-peskine-arm — APPROVED** [review]
I'm happy that the changes are non-harmful. I have not personally checked that they resolve the warnings as advertised, but I am satisfied that [the reporter has reported that they do](https://github.com/Mbed-TLS/mbedtls/issues/6924#issuecomment-1398791958).

**Pull request merged**
